src/commands/junit/upload.ts

```diff
@@ -24,6 +26,20 @@ import {buildPath} from '../../helpers/utils'
 const errorCodesNoRetry = [400, 403, 413]
 const errorCodesStopUpload = [400, 403]
 
+const validateXml = (xmlFilePath: string) => {
+  const xmlFileContentString = String(fs.readFileSync(xmlFilePath))
+  const validationOutput = xmlParser.validate(xmlFileContentString)
+  if (validationOutput !== true) {
+    return validationOutput.err.msg
+  }
+  const xmlFileJSON = xmlParser.parse(String(xmlFileContentString))
```

Comment:
This looks like it parses the full XML. It could get a bit slow for huge reports. Does this library have something to check only the root element?

Reply:
I don't think it does. I've checked alternative libraries and they don't either. We need to go through the whole report for validation anyway, don't we?

Reply:
tested with a rather big file: upload is still a much bigger factor, so I don't think we should worry about it